### PR TITLE
Condense settings page layout into tabs

### PR DESCRIPTION
**Pull request by jayeclark**
Signed-off-by: Jay Clark <jay@jayeclark.dev>

Hi there @fabian-peters! I came across issue #16 and had some ideas on a possible reorg of the settings form. I still need to clean up the code & styles (particularly the inline styling), and tweak a few minor UI issues, but I thought I'd open a draft PR first to check if this is going in the direction you envisioned and claim the issue if so! (See gif below for UI after alterations)

**_Update 10/11:_** Code has been cleaned up and the discussed changes have been made - I think it looks a lot cleaner! I did have to slightly adjust the size of the control window in order to eliminate the need for settings page scrolling, but it shouldn't affect the other functionality as far as I can tell. I also moved 'save timer history' and 'save sub history' to the general settings page, primarily for layout reasons. 

**Initial Draft (old)**
SettingsTabs

**IMPROVED VERSION (V2)**
SettingsTabs-V2

### CODE REVIEW

**Comment by fabian-peters** [comment]
Hi @jayeclark, cool to see you're interested in contributing. First of all, I think it looks great! Feel free to claim the issue.

There are 2 things I noticed:
- I think it would be better if the text/number inputs had the same width
- I don't think it makes sense to hide the "Add'l Time" for higher tiers. In case you're not familiar with Twitch, there are different tiers of subs (higher tiers are more expensive). The idea here is that if someone pays more, more time will be added to the timer. I would just show all 3 fields all the time. If you want to make it fancy, you could build some kind of switch between using the same value for all tiers and individual values for each tier.

One more note, I added the horizontal lines to get at least some kind of structure into the endless list of settings as it was before. I'll leave it up to you if you think they are still needed.

So again, it looks great so far, thank you

**Comment by jayeclark** [comment]
Hi @fabian-peters, thanks for the feedback! 

I’m familiar with Twitch - hiding the Tier 2 and 3 sub times until a Tier 1 time was entered was trying to accomplish the same thing as the switch that you suggested, but I like your idea better. 

Now that I think about it a little more, should ‘additional time tier 1’ even be in there? What’s the use case where the ‘additional time’ added for a Tier 1 sub would be anything other than 0 (wouldn’t the main time be what applies to a Tier 1 sub)?

I will fiddle around with the input sizing to make it more consistent, some of the input field sizes had to change width in order to accommodate the label being all on one line.

That’s also a good idea about eliminating some of the potentially unnecessary horizontal lines.

**Comment by fabian-peters** [comment]
> Now that I think about it a little more, should ‘additional time tier 1’ even be in there? What’s the use case where the ‘additional time’ added for a Tier 1 sub would be anything other than 0 (wouldn’t the main time be what applies to a Tier 1 sub)?

At the moment there is no 'main time' to add. Maybe we should rename the settings to make that clearer.

Initial time is the time the timer starts at. Once the timer is running, this isn't used anymore.
Additional Time Tier 1 is the time that gets added for a tier 1 sub. If it is set to 0, the timer isn't increased for Tier 1 subs.
Additional Time Tier 2/3 is the same but for Tier 2/3 subs. This isn't cumulative. If it is set to 60, 60 seconds are added regardless of the Tier 1 setting.

Hope this clears any confusion regarding those settings.

**Comment by jayeclark** [comment]
Ahhh yes, now I understand. I think the wording could be made a little clearer, maybe ‘initial countdown time’ or something like that for the initial time? Just for the noob setting things up for the first time. I really misinterpreted that part of the settings!

File: src/main.ts
Comment (fabian-peters) [inline src/main.ts:22]:
```suggestion
    height: 400,
```

File: src/main.ts
Comment (fabian-peters) [inline src/main.ts:29]:
```suggestion
    width: 430
```

**Review by fabian-peters — COMMENTED** [review]
The new version looks awesome, thank you for that.
2 things before we can merge this:
- There are still scrollbars on Windows. Height 400 and width 430 worked for me. Are there any styles or something that would need to be adapted to that size?
- I'm happy to list you as contributor if you want to. You can just update the package.json or I will do it after the merge (README will follow sometime in the future)

**Comment by jayeclark** [comment]
Hmm, I have been avoiding making a windows partition on my Mac, but I probably should - this isn't the first time that there has been an odd minor layout difference. 

There shouldn't be anything that would need to be adjusted to accommodate those different dimensions, but let me change the dimensions locally just to double check. If everything looks good I'll add a commit fixing that issue.

And sure, I would love to be listed as a contributor! I'm in the middle of an open source challenge right now (so, I probably will not be contributing more than 2x to any repo until I finish the challenge) but I do intend to come back and contribute again to the projects with responsive maintainers!

**Review by fabian-peters — APPROVED** [review]
Looks good now, from my side it can be merged. You are done, right?
Good luck for your challenge!

**Comment by jayeclark** [comment]
Yes, I believe it’s all set!

**Pull request merged**
